masu/external/downloader/aws/aws_report_downloader.py

```diff
@@ -104,6 +106,52 @@ def manifest_date_format(self):
         """Set the AWS manifest date format."""
         return '%Y%m%dT000000.000Z'
 
+    def _check_size(self, s3key, check_inflate=False):
+        """Check the size of an S3 file.
+
+        Determine if there is enough local space to download and decompress the
+        file.
+
+        Args:
+            s3key (str): the key name of the S3 object to check
+            check_inflate (bool): if the file is compressed, evaluate the file's decompressed size.
+
+        Returns:
+            (bool): whether the file can be safely stored (and decompressed)
+
+        """
+        size_ok = False
+
+        s3fileobj = self.s3_client.get_object(Bucket=self.report.get('S3Bucket'),
+                                              Key=s3key)
+        size = int(s3fileobj.get('ContentLength', -1))
+
+        if size < 1:
+            raise AWSReportDownloaderError(f'Invalid size for S3 object: {s3fileobj}')
+
+        free_space = shutil.disk_usage(self.download_path)[2]
+        if size < free_space:
+            size_ok = True
+
+        LOG.debug('%s is %s bytes; Download path has %s free',
+                  s3key, size, free_space)
+
+        ext = os.path.splitext(s3key)[1]
```

Comment:
Check to make sure extension is present?
```
ext = None
if os.path.splittext(s3key):
    ext = os.path.splitext(s3key)[1]
```

Reply:
Sure. 

I prefer `.get()` to an if-else, but point taken. :)